tempodb/backend/gcs/gcs.go

```diff
@@ -168,7 +173,15 @@ func (rw *readerWriter) Shutdown() {
 }
 
 func (rw *readerWriter) writer(ctx context.Context, name string) *storage.Writer {
-	w := rw.bucket.Object(name).NewWriter(ctx)
+	o := rw.bucket.Object(name)
+	if configuredObjectAttributes(rw.cfg.ObjAttrs) {
+		_, err := o.Update(ctx, rw.cfg.ObjAttrs)
+		if err != nil {
+			level.Error(rw.logger).Log("msg", "failed to update object attributes", "objectName", name, "err", err)
```

Comment:
I think this could lead to a nil reference/panic in `.Append()` and `.Write()`.  Thoughts on returning the error and doing normal error checking?

Reply:
Good call, I didn't check those methods, but I like returning an error better than a nil that the caller has to know something about.  That said, should we also update the `.Append()` and `.Write()` methods to avoid a nil pointer should one get returned, or just rely on the error value to tell us we aren't in a good state?

Reply:
I think relying on the error is idiomatic go and ok.